### PR TITLE
Delete Useless options in controller

### PR DESCRIPTION
**Pull request by czybjtu**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/scheduler-plugins/issues/567

#### Does this PR introduce a user-facing change?

```release-note
Delete kubeConfig/incluster/masterUrl options in controller. 
Implentment --worker option.
```

### CODE REVIEW

**Comment by zwpaper** [comment]
hi @czybjtu, thanks for the PR, but maybe we should implement those options instead of deleting them 

WDYT @Huang-Wei ?

File: cmd/controller/app/server.go [hunk 1]
```diff
@@ -129,53 +86,9 @@ func Run(s *ServerRunOptions) error {
...
-		if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-			setupLog.Error(err, "unable to start manager")
-			panic(err)
-		}
-
-		select {}
-	}
-	schedInformerFactory.Start(stopCh)
-	coreInformerFactory.Start(stopCh)
-	if !s.EnableLeaderElection {
```
Comment (Huang-Wei) [inline cmd/controller/app/server.go:143]: this seems to be totally disabling leader election?

**Comment by Huang-Wei** [comment]
> hi @czybjtu, thanks for the PR, but maybe we should implement those options instead of deleting them

Yes, we need to bring the functionality back, which is part of the migration; otherwise, it's a breaking change as params like workers, leader elections are not supported.

@czybjtu are you going to remove and add the param/function back, via 2 PRs? If so, I'm leaning towards merging into one PR, so reviewers only need to review function changes while keep parameters unchanged (or with minimum changes).

File: cmd/controller/app/server.go (hunk 1 above)
Comment (czybjtu) [reply cmd/controller/app/server.go:143]: No need to implement this manually, the ctrl runtime can handle this instead. We could enable leader election when create `ctrl.NewManager`

**Comment by czybjtu** [comment]
AFAIK, after we migrate controllers to controller-runtime, there is no need to implement `kubeConfig/incluster/masterUrl`  manually. The controller-runtime can handle this instead.
The `--wokers` option was never be implemented before, So I delete this option in this pr. Maybe it's better to implement it. I'll take a look later.

File: cmd/controller/app/server.go (hunk 1 above)
Comment (czybjtu) [reply cmd/controller/app/server.go:143]: https://github.com/kubernetes-sigs/scheduler-plugins/blob/aeb7ff27f5fa6559c588d7a133fe933d1fc65ca6/cmd/controller/app/server.go#L93-L100

**Comment by zwpaper** [comment]
hi @czybjtu, sorry for the late reply again, I was kinda busy recently.

my concern is that:
1. ctrl-runtime automatically added a kubeconfig command line flag to the controller, it may be worth a comment in the Option struct.
2. the `kubeconfig` cmd flag was changed from `kubeConfig`, we should add this breaking change in the release note @Huang-Wei 
3. it seems that ctrl-runtime did not handle the `masterAddr` flag, and it also seems can be replaced by the `kubeconfig` flag, we may also drop it and notice it in the release note?
4. the `inCluster` flag seems to be reasonable to drop, as it can be detected by whether `kubeconfig` is specified.

ps, I only glanced at the PR, will try to find some free cycles to check it deeply.

**Comment by Huang-Wei** [comment]
/ok-to-test

I'm not concerned with the features that didn't exist before, we can keep the behavior as-is as long as we can confirm some parameters didn't work (like masters).

For kubeconfig -> kubeConfig and changes alike, I'm less concerned as well. Let's just stick to ctrl-runtime's naming conversion and write a release note about that.

File: pkg/controllers/elasticquota_controller.go [hunk 2]
```diff
@@ -173,5 +175,6 @@ func (r *ElasticQuotaReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		Watches(&source.Kind{Type: &v1.Pod{}}, &handler.EnqueueRequestForObject{}).
 		For(&schedv1alpha1.ElasticQuota{}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.Workers}).
```
Comment (Huang-Wei) [inline pkg/controllers/elasticquota_controller.go:178]: thanks for adding this.

**Comment by Huang-Wei** [comment]
/approve

Leaving /lgtm to @zwpaper .

**Comment by zwpaper** [comment]
/lgtm

thanks for the contribution @czybjtu!

**Pull request merged**